packages/components/src/components/hds/app-header/index.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+export interface HdsAppHeaderSignature {
+  Blocks: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    globalBefore?: any;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    logo: any;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    globalAfter?: any;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Used "any" type for now for these. Should these have more specific types?

(The logo named block is intended to only accept the HomeLink component so I may update that next.)

Reply:
I tried changing the logo block to only accept the HomeLink component.

Example: `Logo: ComponentLike<HdsAppHeaderHomeLinkSignature>;`

Which was fine but then I couldn't get it to work within the hbs template. I think because it won't work together with using a named block.